packages/astro/src/compiler/codegen/index.ts

```diff
@@ -467,10 +467,17 @@ async function compileHtml(enterNode: TemplateNode, state: CodegenState, compile
             const children: string[] = await Promise.all((node.children ?? []).map((child) => compileHtml(child, state, compileOptions)));
             let raw = '';
             let nextChildIndex = 0;
+            const shouldUseFragment = children.length > 1 && (node.codeChunks[0].match(/\=\>/g) || node.codeChunks[0].match(/\bfunction\b/g)) && !(node.codeChunks[0].match(/\?/g) && node.codeChunks[1].match(/\:/g));
```

Comment:
Can you add a comment on what this is looking for?

If possible, it would be nice to combine this regexes.

Reply:
Replaced this implementation with a more robust one that just checks if the last character is `(` or the first character is `)`